### PR TITLE
Add system tests for Vertex AI operators in new approach

### PR DESCRIPTION
**Pull request by MaksYermak**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by lwyszomi — APPROVED** [review]

**Comment by MaksYermak** [comment]
> Those are really huge csv files in the resources - it will not pass PR in AIrflow ;) Will you replace it somehow? Or we should think about finally implementing solution for storing the files in the external cloud.

Yes of course I will try to compress it and decompress it in test. Also Bartek I have a question to you, how can I run this bunch of tests in a better way:
```
# Needed to run the example DAG with pytest (see: tests/system/README.md#run_via_pytest)
custom_container_test_run = get_test_run(custom_container_dag)
python_package_test_run = get_test_run(python_package_dag)
custom_test_run = get_test_run(custom_dag)
list_custom_test_run = get_test_run(list_custom_job_dag)
```
for our `pytest`?

**Comment by bhirsz** [comment]
> > Those are really huge csv files in the resources - it will not pass PR in AIrflow ;) Will you replace it somehow? Or we should think about finally implementing solution for storing the files in the external cloud.
> 
> Yes of course I will try to compress it and decompress it in test. Also Bartek I have a question to you, how can I run this bunch of tests in a better way:
> 
> ```
> # Needed to run the example DAG with pytest (see: tests/system/README.md#run_via_pytest)
> custom_container_test_run = get_test_run(custom_container_dag)
> python_package_test_run = get_test_run(python_package_dag)
> custom_test_run = get_test_run(custom_dag)
> list_custom_test_run = get_test_run(list_custom_job_dag)
> ```
> 
> for our `pytest`?

You mean you want to run several tests? You can point to the whole directory, or several files:
```
pytest --system google tests/system/providers/google/cloud/your/directory
```

**Comment by MaksYermak** [comment]
> > > Those are really huge csv files in the resources - it will not pass PR in AIrflow ;) Will you replace it somehow? Or we should think about finally implementing solution for storing the files in the external cloud.
> > 
> > 
> > Yes of course I will try to compress it and decompress it in test. Also Bartek I have a question to you, how can I run this bunch of tests in a better way:
> > ```
> > # Needed to run the example DAG with pytest (see: tests/system/README.md#run_via_pytest)
> > custom_container_test_run = get_test_run(custom_container_dag)
> > python_package_test_run = get_test_run(python_package_dag)
> > custom_test_run = get_test_run(custom_dag)
> > list_custom_test_run = get_test_run(list_custom_job_dag)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > for our `pytest`?
> 
> You mean you want to run several tests? You can point to the whole directory, or several files:
> 
> ```
> pytest --system google tests/system/providers/google/cloud/your/directory
> ```

@bhirsz no, I mean how I should use `get_test_run` function in case when file has several DAGs inside?

**Comment by bhirsz** [comment]
> @bhirsz no, I mean how I should use `get_test_run` function in case when file has several DAGs inside?

To be honest, I have no idea :) It is designed to support one dag per one file (which usually makes it more readable). It's actually common practise in testing to prefer DAMP over DRY - better to repeat some of the code defining separate example DAG than keep everything in one place.

File: tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py [hunk 1]
```diff
@@ -0,0 +1,688 @@
...
+IMAGE_GCS_BUCKET_NAME = f"bucket_image_{DAG_ID}_{ENV_ID}"
+TABULAR_GCS_BUCKET_NAME = f"bucket_tabular_{DAG_ID}_{ENV_ID}"
+TEXT_GCS_BUCKET_NAME = f"bucket_text_{DAG_ID}_{ENV_ID}"
+VIDEO_GCS_BUCKET_NAME = f"bucket_custom_python_{DAG_ID}_{ENV_ID}"
+
+FORECAST_ZIP_CSV_FILE_LOCAL_PATH = str(Path(__file__).parent / "resources" / "forecast-dataset.csv.zip")
+IMAGE_ZIP_CSV_FILE_LOCAL_PATH = str(Path(__file__).parent / "resources" / "image-dataset.csv.zip")
+TABULAR_ZIP_CSV_FILE_LOCAL_PATH = str(Path(__file__).parent / "resources" / "tabular-dataset.csv.zip")
+TEXT_ZIP_CSV_FILE_LOCAL_PATH = str(Path(__file__).parent / "resources" / "text-dataset.csv.zip")
+VIDEO_ZIP_CSV_FILE_LOCAL_PATH = str(Path(__file__).parent / "resources" / "video-dataset.csv.zip")
```
Comment (bhirsz) [inline tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py:68]: I know it's not your change (@VladaZakharova ) but could you use one common variable for resources path:
```suggestion
RESOURCES_PATH = Path(__file__).parent / "resources"
FORECAST_ZIP_CSV_FILE_LOCAL_PATH = str(RESOURCES_PATH / "forecast-dataset.csv.zip")
IMAGE_ZIP_CSV_FILE_LOCAL_PATH = str(RESOURCES_PATH / "image-dataset.csv.zip")
TABULAR_ZIP_CSV_FILE_LOCAL_PATH = str(RESOURCES_PATH / "tabular-dataset.csv.zip")
TEXT_ZIP_CSV_FILE_LOCAL_PATH = str(RESOURCES_PATH / "text-dataset.csv.zip")
VIDEO_ZIP_CSV_FILE_LOCAL_PATH = str(RESOURCES_PATH / "video-dataset.csv.zip")
```

File: tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py [hunk 2]
```diff
@@ -0,0 +1,688 @@
...
+        dataset=FORECAST_DATASET,
+        region=REGION,
+        project_id=PROJECT_ID,
+    )
+    forecast_dataset_id = create_forecast_dataset.output['dataset_id']
+
+    # [START how_to_cloud_vertex_ai_create_auto_ml_forecasting_training_job_operator]
+    create_auto_ml_forecasting_training_job = CreateAutoMLForecastingTrainingJobOperator(
+        task_id="auto_ml_forecasting_task",
+        display_name=f"auto-ml-forecasting-{ENV_ID}",
```
Comment (bhirsz) [inline tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py:207]: To not confuse users with "ENV_ID" in the example, we could introduce variable for it (and it's used in two places btw):

```
FORECASTING_DISPLAY_NAME = f"auto-ml-forecasting-{ENV_ID}"
```
Same for  model_display_name=f"auto-ml-forecasting-model-{ENV_ID}"

File: tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py [hunk 3]
```diff
@@ -0,0 +1,688 @@
...
+        >> delete_auto_ml_forecasting_training_job
+        >> delete_forecast_dataset
+        >> delete_bucket
+        >> clear_folder
+    )
+
+
+with models.DAG(
+    f"{DAG_ID}_image_training_job",
+    schedule_interval="@once",
```
Comment (bhirsz) [inline tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py:274]: schedule_interval -> schedule for all DAGs

File: tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py [hunk 4]
```diff
@@ -0,0 +1,688 @@
...
+        region=REGION,
+        project_id=PROJECT_ID,
+    )
+    # [END how_to_cloud_vertex_ai_create_auto_ml_image_training_job_operator]
+
+    delete_auto_ml_image_training_job = DeleteAutoMLTrainingJobOperator(
+        task_id="delete_auto_ml_training_job",
+        training_pipeline_id=create_auto_ml_image_training_job.output['training_id'],
+        region=REGION,
+        project_id=PROJECT_ID,
```
Comment (bhirsz) [inline tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py:336]: Do we need trigger_rule for it or DeleteDatasetOperator removes the whole resource anyway? (I don't know the service specifics)

Comment (VladaZakharova) [reply tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py:336]: Yes, it is needed. I have added this trigger_rule to all other delete tasks as well :)

File: tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py (hunk 3 above)
Comment (VladaZakharova) [reply tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py:274]: Done

File: tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py (hunk 2 above)
Comment (VladaZakharova) [reply tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py:207]: Done

File: tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py (hunk 1 above)
Comment (VladaZakharova) [reply tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py:68]: It is really good idea, thanks!

**Comment by potiuk** [comment]
Needs conflict resolving / rebase - also it has more changes than just converting to system tests (adding new links and handling missing ids. Can we split-off the new changes to separate PR please ? Then it will also be visible in changelog.

**Comment by potiuk** [comment]
Can you please rebase @MaksYermak ? There are some depdency changes and I want to make sure the CI checks are run fully.

**Pull request closed without merging**